tests/test_vtp.rb

```diff
@@ -77,57 +71,42 @@ def test_vtp_domain_enable_disable
     assert_empty(Vtp.domain)
   end
 
-  def test_vtp_create_valid
-    vtp_domain('accounting')
-    s = @device.cmd("show run vtp | incl '^vtp domain'")
-    assert_match(/^vtp domain accounting/, s,
-                 'Error: failed to create vtp domain')
+  def test_create_valid
+    vtp = vtp_domain('accounting')
+    assert_equal('accounting', vtp.domain)
   end
 
-  def test_vtp_domain_name_change
+  def test_domain_name_change
     vtp = vtp_domain('accounting')
     vtp_domain('uplink')
     assert_equal('uplink', vtp.domain,
                  'Error: vtp domain name incorrect')
   end
 
-  def test_vtp_create_preconfig_no_change
-    config('feature vtp', 'vtp domain accounting')
-
-    vtp = vtp_domain('accounting')
-    assert_equal('accounting', vtp.domain,
-                 'Error: vtp domain wrong')
-  end
-
-  def test_vtp_create_double
-    vtp = vtp_domain('accounting')
-    vtp_new = vtp_domain('accounting')
-
-    assert_equal('accounting', vtp.domain,
-                 'Error: vtp domain wrong')
-    assert_equal('accounting', vtp_new.domain,
-                 'Error: vtp_new domain wrong')
+  def test_create_double
+    vtp_domain('accounting')
+    assert_raises(Cisco::CliError) { vtp_domain('accounting') }
   end
 
-  def test_vtp_create_domain_invalid
+  def test_create_domain_invalid
     assert_raises(ArgumentError) do
       vtp_domain(node)
     end
   end
 
-  def test_vtp_create_domain_nil
+  def test_create_domain_nil
     assert_raises(ArgumentError) do
       vtp_domain(nil)
     end
   end
 
-  def test_vtp_create_domain_empty
+  def test_create_domain_empty
     assert_raises(ArgumentError) do
       vtp_domain('')
     end
   end
```

Comment:
I am usually in favor of a higher level of test granularity so that we don't hold a large set of test cases hostage if one of them fails but in this case I'll allow it :)